Test/Unit/Controller/Order/SaveTest.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+namespace Bolt\Boltpay\Test\Unit\Controller\Order;
+
+use Bolt\Boltpay\Controller\Order\Save;
+use Bolt\Boltpay\Helper\Bugsnag;
+use Bolt\Boltpay\Helper\Config;
+use Bolt\Boltpay\Helper\Order as OrderHelper;
+use Magento\Checkout\Model\Session;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\App\Test\Unit\ObjectManagerFactoryTest;
+use Magento\Framework\Controller\Result\JsonFactory as ResultJsonFactory;
+use Magento\Framework\Controller\Result\Json;
+use Magento\Framework\DataObjectFactory;
+use Magento\Framework\UrlInterface;
+use Magento\Sales\Model\Order;
+use Magento\Quote\Model\Quote;
+use PHPUnit\Framework\TestCase;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use PHPUnit_Framework_MockObject_MockObject as MockObject;
+
+class SaveTest extends TestCase
+{
+    const ORDER_ID = 1234;
+    const QUOTE_ID = 5678;
+    const INCREMENT_ID = 1235;
+    const STATUS = "Ready";
+    const REFERENCE = "referenceValue";
+    const SUCCESS_URL = "http://url.return.value/";
+    const EXCEPTION_MESSAGE = "Exception Message";
+
+    /**
+     * @var Order | MockObject
+     */
+    private $orderMock;
+
+    /**
+     * @var Quote | MockObject
+     */
+    private $quoteMock;
+
+    /**
+     * @var Bugsnag | MockObject
+     */
+    private $bugsnagMock;
+
+    /**
+     * @var OrderHelper | MockObject
+     */
+    private $orderHelper;
+
+    /**
+     * @var Session | MockObject
+     */
+    private $checkoutSession;
+
+    /**
+     * @var Config | MockObject
+     */
+    private $configHelper;
+
+    /**
+     * @var DataObjectFactory
+     */
+    private $dataObjectFactory;
+
+    /**
+     * @var Context
+     */
+    private $context;
+
+    protected function setUp()
+    {
+        $this->initRequiredMocks();
+    }
+
+    public function testExecute_HappyPath()
+    {
+        $result = [
+            'status' => 'success',
+            'success_url' => self::SUCCESS_URL
+        ];
+
+        $json = $this->getMockBuilder(Json::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $json->expects($this->once())
+            ->method('setData')
+            ->with($result);
+        $jsonFactory = $this->getMockBuilder(ResultJsonFactory::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $jsonFactory->method('create')->willReturn($json);
+
+        $url = $this->createMock(UrlInterface::class);
+        $url->method('getUrl')->willReturn(self::SUCCESS_URL);
+        $this->context->method('getUrl')->willReturn($url);
+
+        $this->configHelper->method('getSuccessPageRedirect');
+
```

Comment:
This seems unfinished. You should probably set a return value here (checkout/onepage/success ?) and set the same as the expectation to the `$url->method('getUrl')` above.

Reply:
The method is actually configured in the init method, its presence here was an un-removed artifact of writing this. Removing it.